### PR TITLE
Serve SaltGUI from different host then salt master

### LINKED ISSUE #238: Serve SaltGUI from different host then salt master
**Is your feature request related to a problem? Please describe.**
It would be nice if we could configure saltgui on different host then salt master/api.

So basicaly I would like to decauple ui (and serve it from nginx) from salt master/api. This of course should not be mandatory.

**Describe the solution you'd like**
A way to inject config like `apiurl` which will point to proper salt master/api endpoint.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by dawidmalina**
closes #238

### CODE REVIEW

**Comment by dawidmalina** [comment]
To make it work you need to configure: 
```
API_URL: '/api'
```

> Production environments should use certificates and https connections configured on nginx side!

**Comment by dawidmalina** [comment]
@erwindon can you give early feedback?

**Comment by erwindon** [comment]
That looks very reasonable! sorry, had no time to test it, see also the list of other work :-)

A small chapter in `/README.md` should be added. Just after `#Pillars`? Title is "#Separate SaltGUI host". explaining why someone would want this; what needs to be configured; and what also needs to be setup (nginx).

please fix config.js, var --> const

**Comment by erwindon** [comment]
Can you please rebase the branch?
The eslint error that appears has been fixed in `master`.

File: README.md [hunk 1]
```diff
@@ -164,6 +164,47 @@ This is then the built-in version from the brower.
 Typical effect is that it is shown slightly delayed and that is looks a bit primitive.
 The only other allowed value is "none", with the effect that no tooltips are shown at all.
 
+## Separate SaltGUI host
+In some specific environments you might not be able to server SaltGUI directly from salt-api.
```
Comment (erwindon) [inline README.md]: server -> serve

File: README.md [hunk 2]
```diff
@@ -164,6 +164,47 @@ This is then the built-in version from the brower.
...
+In this case you might want to configure web server (for example NGINX) to server SaltGui
```
Comment (erwindon) [inline README.md]: this->that
to configure web server -> to configure a web server
to server -> to serve

File: README.md [hunk 3]
```diff
@@ -164,6 +164,47 @@ This is then the built-in version from the brower.
 Typical effect is that it is shown slightly delayed and that is looks a bit primitive.
 The only other allowed value is "none", with the effect that no tooltips are shown at all.
 
+## Separate SaltGUI host
+In some specific environments you might not be able to server SaltGUI directly from salt-api.
+In this case you might want to configure web server (for example NGINX) to server SaltGui 
+and use it as proxy to salt-api such that requests are answered from the same origin from the browser point of view.
```
Comment (erwindon) [inline README.md]: such -> so

File: README.md [hunk 4]
```diff
@@ -164,6 +164,47 @@ This is then the built-in version from the brower.
...
+
+Simple NGINX configuration might looks like this:
```
Comment (erwindon) [inline README.md]: Simple or Sample?
looks -> look

File: README.md [hunk 5]
```diff
@@ -164,6 +164,47 @@ This is then the built-in version from the brower.
...
+```
+
+The value of the `API_URL` in the `config.js` file shall point to path where salt-api is exposed. 
+```
+const config = {
+  API_URL: '/api'
+};
+```
+
+> Currenlty you can't use totally independend Salt-api without proxy as support for CORS preflight request is not properly support.
```
Comment (erwindon) [inline README.md]: independend --> independent

File: docker/docker-compose.yml [hunk 6]
```diff
@@ -16,19 +16,31 @@ services:
...
 
   saltminion-centos:
     image: erwindon/saltgui-saltminion-centos:2019.2.0
     hostname: saltminion-centos
     depends_on:
-       - saltmaster-local
+      - saltmaster-local
     restart: on-failure
+
+  # This example will demonstrate how to use nginx for saltgui and proxy to salt-api
```
Comment (erwindon) [inline docker/docker-compose.yml]: saltgui -> SaltGUI

File: saltgui/static/scripts/config.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+const config = {
+  // API_URL: '/api'
```
Comment (erwindon) [inline saltgui/static/scripts/config.js]: or 'ht<i></i>tp://hostname:port'

File: saltgui/static/scripts/Api.js [hunk 8]
```diff
@@ -7,8 +7,8 @@ export class HTTPError extends Error {
 }
 
 export class API {
-  constructor(router, apiurl="") {
-    this.APIURL = apiurl;
+  constructor(router) {
+    this.APIURL = config.API_URL;
```
Comment (erwindon) [inline saltgui/static/scripts/Api.js]: can this assignment be removed?
and let function `apiRequest` directly use `config.API_URL`

File: saltgui/static/scripts/config.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
...
+  API_URL: ''
+};
```
Comment (erwindon) [inline saltgui/static/scripts/config.js]: can you please save the file with a LF on the final line?

**Comment by erwindon** [comment]
@dawidmalina 
When can you make the requested changes?

**Comment by dawidmalina** [comment]
I will fix all issues in the comming days.

File: saltgui/static/scripts/config.js (hunk 7 above)
Comment (dawidmalina) [reply saltgui/static/scripts/config.js]: The problem here is that we can't use fully independent salt-api as mentioned in the main readme.

**Comment by erwindon** [comment]
@dawidmalina 
Before I merge this, a question: are you using this in any production-like situation?
If not, how would you describe the situation in just a few words?

**Comment by dawidmalina** [comment]
This is not yet deployed in any production-like environments. The main blocker in my case are the network flows. I don't want to open flows from client facing network to salt-master (salt-api). The only flow I would like to have is client - nginx (reverse proxy) and then (internally) nginx - salt-master (api). In addition having this separated allows me to simplify saltgui maintanace as I can run it as docker container.

**Pull request merged**